src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp

```diff
@@ -417,6 +417,8 @@ SIZE NonClientIslandWindow::GetTotalNonClientExclusiveSize(UINT dpi) const noexc
 
     islandFrame.top = -topBorderVisibleHeight;
 
+    // If we have a titlebar, this is being called after we've initialized, and
+    // we can just ask that titlebar how big it wants to be.
     const auto titleBarHeight = _titlebar ? static_cast<LONG>(_titlebar.ActualHeight()) : 0;

```

Comment:
Isn't it adding the title bar twice now: Here and in `AppLogic::GetLaunchDimensions`? But this works because it's not initialized yet so it adds `0`.

Reply:
Thanks for filing that! You're right, that method name is definitely confusing. IIRC, in the PR that introduced it there was some back and forth on the naming but we couldn't come up with a better name. With this PR, it should be a little easier to at least solve that bug (#4827).